Calibration/PCCAlCaRecoProducers/interface/AlcaPCCProducer.h

```diff
@@ -0,0 +1,55 @@
+#ifndef Calibration_PCCAlCaRecoProducers_AlcaPCCProducer_h
+#define Calibration_PCCAlCaRecoProducers_AlcaPCCProducer_h
+
+/**_________________________________________________________________
+   class:   AlcaPCCProducer.h
+   package: Calibration/PCCAlCaRecoProducers
+   
+
+ authors:Sam Higginbotham (shigginb@cern.ch) and Chris Palmer (capalmer@cern.ch) 
+
+________________________________________________________________**/
+
+
+// C++ standard
+#include <string>
+// CMS
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDProducer.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "DataFormats/Luminosity/interface/PCC.h"
+
+class AlcaPCCProducer : public edm::one::EDProducer<edm::EndLuminosityBlockProducer,
```

Comment:
Probably... from the twiki once up from these I see:

A Stream module is one which only cares about Events individually and not as a group and therefore works just fine if it only sees a subset of the Events coming from the Source.

and 

A Global module is one which must see all Events coming from a Source in order to do its job. 

In our case each event is processed individually but only saves at the end of a lumi section (or series of lumi sections). So conceptually I don't see where this producer fits most appropriately.  Maybe you could educate us about which type would be best for this producer and highlight the benefits for it?